_el/exercises.md

```diff
@@ -336,172 +204,52 @@ piece of code with the output it produces".
+
+Οι [[Srid2016](#CITE)] συζητούν στρατηγικές ανταλλαγής σχολίων με τους μαθητές κατά τη διάρκεια της αυτόματης βαθμολόγησής τους. Μια πρώτη στρατηγική είναι να δίνεται το αναμενόμενο αποτέλεσμα των δοκιμών -- αλλά στη συνέχεια οι μαθητές μπορούν να ενσωματώσουν αυτό το αποτέλεσμα στον κώδικά τους (γιατί ότι μπορεί να λειτουργήσει ως εύκολη λύση, θα γίνει). Μια εναλλακτική είναι η αναφορά των αποτελεσμάτων επιτυχίας / αποτυχίας για τον κώδικα των εκπαιδευομένων, αλλά να δίνονται τα πραγματικά αποτελέσματα των δοκιμνών (ως είσοδος / έξοδος του προγράμματος) μετά την καταλληκτική ημερομηνία υποβολής. Αυτό μπορεί να είναι απογοητευτικό, διότι τους λέει στους μαθητές ότι είναι λάθος, αλλά όχι το γιατί είναι.
+
+Μια τρίτη επιλογή είναι να χρησιμοποιήσετε μια τεχνική που ονομάζεται [κατακερματισμός](#g:hashing) για τη δημιουργία μιας τιμής που εξαρτάται από το πραγματικό αποτέλεσμα αλλά δεν το αποκαλύπτει. Αν ο χρήστης παράγει ακριβώς την ίδια έξοδο, το κλειδί κατακερματισμού θα είναι το ίδιο με το κλειδί κατακερματισμού της σωστής εξόδου, και η οποία θα ξεκλειδώσει τη λύση, αλλά είναι αδύνατο να εργαστεί κανείς αντίστροφα και από το κλειδί κατακερματισμού να καταλάβει ποιά θα πρέπει να είναι πραγματικά η έξοδος. Ο κατακερματισμός χρησιμοποιείται για τη δημιουργία ψηφιακών υπογραφών για έγγραφα και απαιτεί λίγο περισσότερη δουλειά και εξήγηση για τη δημιουργία του, αλλά επιτυγχάνει μια καλή ισορροπία μεταξύ της πρόωρης αποκάλυψης των απαντήσεων με την αποκάλυψή τους όταν πραγματική μπορεί να βοηθήσει.
+
+## Συλλογιστική Υψηλού Επιπέδου {#s:exercises-higher}
+
+Υπάρχουν πολλοί άλλοι τύποι ασκήσεων προγραμματισμού ο οποίοι είναι εξίσου δύσκολο να αξιολογηθούν τόσο για τους εκπαιδευτικούς σε μια τάξη μερικών δεκάδων μαθητών και όσο και για τις αυτοματοποιημένες πλατφόρμες βαθμολόγησης. Μεγαλύτερα εργασίες προγραμματισμού, ή προγράμματα στα οποία οι μαθητές θέτουν τους δικούς τους στόχους, (ελπίζουμε ότι) είναι η γενικότερη κατεύθυνση στην οποίο πηγαίνουν τα μαθήματα προγραμματισμού. Οι συζητήσεις ελεύθερης μορφής ή η αιφνίδια συγγραφή κώδικα (twitch-coding) ([s:performance-live](#SECTION)) είναι επίσης πολύτιμες μορφές ασκήσεων, αλλά δεν μπορούν να γίνουν σε μεγάλη κλίμακα.
+
+Η *Αναθεώρηση Κώδικα*, από την άλλη πλευρά, είναι δύσκολο να βαθμολογηθεί αυτόματα στη γενική περίπτωση, αλλά μπορεί να αντιμετωπιστεί εάν στους μαθητές δοθούν συγκεκριμένες οδηγίες (π.χ. λίστα των σφαλμάτων που πρέπει να αναζητήσετε) και τους ζητθεί να αντιστοιχίσουν συγκεκριμένα σχόλια έναντι συγκεκριμένων γραμμών κώδικα. Για παράδειγμα, ο εκπαιδευόμενος μπορεί να πει ότι υπάρχουν δύο σφάλματα εσοχής και ένα όνομα κακής μεταβλητής, τα οποία και επισημάνει - μια πιο προηγμένη μορφή της άσκησης, θα μπορούσε να είναι μια λίστα από είκοσι έξι είδη παρατηρήσεων που θα μπορούσε να γίνουν σχετικά με τον κώδικα χωρίς
+καθοδήγηση σχετικά με το πόσες από αυτές θα βρουν οι μαθητές.
```

Comment:
θα βρουν--> θα πρέπει να βρουν

Reply:
Τhanks @pdelias ! Fixed in a new commit.